pkg/gadgets/run/tracer/tracer.go

```diff
@@ -566,6 +595,97 @@ func (t *Tracer) runIter(gadgetCtx gadgets.GadgetContext) error {
 	return nil
 }
 
+func (t *Tracer) nextStats(gadgetCtx gadgets.GadgetContext) ([]*types.Event, error) {
+	stats := []*types.Event{}
+	entries := t.statsMap
+
+	defer func() {
+		// Delete elements. TODO: We should ensure to delete only the elements
+		// we read to avoid deleting elements that are not read yet.
+		key, err := entries.NextKeyBytes(nil)
+		if err != nil {
+			gadgetCtx.Logger().Warnf("couldn't get first key to delete: %v", err)
+			return
+		}
+		if key == nil {
+			// Map is empty
+			return
+		}
+
+		for {
+			if err := entries.Delete(key); err != nil {
+				gadgetCtx.Logger().Warnf("couldn't delete value from key: %v", err)
+				return
+			}
+			key, err = entries.NextKeyBytes(key)
+			if err != nil {
+				return
+			}
+			if key == nil {
+				// No more keys
+				break
+			}
+		}
+	}()
+
+	// Gather elements: Start by getting the first key
+	key, err := entries.NextKeyBytes(nil)
+	if err != nil {
+		return nil, fmt.Errorf("getting fist key: %w", err)
+	}
+	if key == nil {
+		// Map is empty
+		return stats, nil
+	}
+
+	// Now iterate over all keys
+	cb := t.processEventFunc(gadgetCtx)
+	for {
+		var rawStat []byte
+		rawStat, err := entries.LookupBytes(key)
+		if err != nil {
+			return nil, fmt.Errorf("looking up value from key: %w", err)
+		}
+
+		stats = append(stats, cb(rawStat))
+
+		key, err = entries.NextKeyBytes(key)
+		if err != nil {
+			return nil, fmt.Errorf("getting next key: %w", err)
+		}
+		if key == nil {
+			// No more keys
+			break
+		}
+	}
+
+	// TODO: How can we sort? Data is still in a raw format by this point.
+	// top.SortStats(stats, t.config.SortBy, &t.colMap)
```

Comment:
I think it's needed to create an instance of columns and keep it on the server side, then use `columnssort.SortEntries`. The easiest way seems to make `getColumns` a function that can be called from run.go and tracer.go, then create the columns when the gadget is run. However, many of that logic will be changed by https://github.com/inspektor-gadget/inspektor-gadget/issues/2131.

Reply:
@flyth, any suggestion here? I still need to handle this. I suppose it will have the same problem after https://github.com/inspektor-gadget/inspektor-gadget/pull/2512, right?